File: driver_cpp/ElevInterface.cpp
#include "ElevInterface.hpp"
#include "channels.hpp"
#include "io.hpp"

#include <assert.h>
#include <stdlib.h>


ElevInterface::ElevInterface() {
    int init_success = io_init();
    assert(init_success && "Unable to initialize elevator hardware!");

    
}

void ElevInterface::set_motor_direction(elev_motor_direction_t dirn) {
    if (dirn == 0){
        io_write_analog(MOTOR, 0);
    } else if (dirn > 0) {
        io_clear_bit(MOTORDIR);
        io_write_analog(MOTOR, MOTOR_SPEED);
    } else if (dirn < 0) {
        io_set_bit(MOTORDIR);
        io_write_analog(MOTOR, MOTOR_SPEED);
    }
}

int ElevInterface::get_floor_sensor_signal(void) {
    if (io_read_bit(SENSOR_FLOOR1)) {
        return 0;
    } else if (io_read_bit(SENSOR_FLOOR2)) {
        return 1;
    } else if (io_read_bit(SENSOR_FLOOR3)) {
        return 2;
    } else if (io_read_bit(SENSOR_FLOOR4)) {
        return 3;
    } else {
        return -1;
    }
}

File: driver_python/ElevInterface.py
import comedi as c
import io as io
import channels as channel
import constants as const

class ElevInterface:
    
    def set_motor_direction(dirn):
    	if dirn == 0:
    		io.write_analog(channel.MOTOR, 0)
    	elif dirn > 0:
    		io.clear_bit(channel.MOTORDIR)
    		io.write_analog(channel.MOTOR, const.MOTOR_SPEED)
    	else:
    		io.set_bit(channel.MOTORDIR)
    		io.write_analog(channel.MOTOR, const.MOTOR_SPEED)

	
	def get_floor_sensor_signal():
		if io.read_bit(channel.SENSOR_FLOOR1):
			return 0
		elif io.read_bit(channel.SENSOR_FLOOR2):
			return 1
		elif io.read_bit(channel.SENSOR_FLOOR3):
			return 2
		elif io.read_bit(channel.SENSOR_FLOOR4):
			return 3
		else:
			return -1


    def __init__(self):
    	init_success = io.init()
    	assert init_success, "Unable to initialize elevator hardware!"

